build/safetyHistogram.js

```diff
@@ -187,46 +190,44 @@ var safetyHistogram = function (webcharts, d3$1) {
     };
 
     function onLayout() {
-        var context = this;
+        var chart = this;
 
         //Add population count container.
         this.controls.wrap.append('div').attr('id', 'populationCount').style('font-style', 'italic');
 
         //Add footnote.
         this.wrap.insert('p', '.wc-chart').attr('class', 'annote').text('Click a bar for details.');
-
-        //Add control to hide or display normal range(s).
-        if (this.config.normal_range) {
-            var normalRange = this.controls.wrap.append('div').attr('id', 'NRcheckbox').style('margin', '.5em').text('Normal Range').append('input').attr('type', 'checkbox');
-            normalRange.on('change', function () {
-                context.wrap.selectAll('.normalRange').attr('visibility', d3.select(this).property('checked') ? 'visible' : 'hidden');
-            });
-        }
     }
 
     function onPreprocess() {
-        var context = this;
+        var _this = this;
 
-        //Capture currently selected filters.
-        var filterSettings = [];
-        var filters = d3.selectAll('.wc-controls .changer').each(function (d) {
-            filterSettings.push({ value_col: d.value_col,
-                value: d3.select(this).selectAll('option').filter(function (d1) {
-                    return d3.select(this).property('selected');
-                }).property('value') });
-        });
-        //Filter data based on currently selected filters.
-        var filtered_data = this.raw_data.filter(function (d) {
-            var match = true;
-            filterSettings.forEach(function (d1) {
-                if (match === true) match = d[d1.value_col] === d1.value || d1.value === 'All';
-            });
-            return match;
+        var chart = this;
+
+        //Filter raw data on currently selected measure.
+        var measure = this.filters.filter(function (filter) {
+            return filter.col === _this.config.measure_col;
+        })[0].val;
+        this.measure_data = this.raw_data.filter(function (d) {
+            return d[_this.config.measure_col] === measure;
         });
-        //Set x-domain based on currently filtered data.
-        this.config.x.domain = d3.extent(filtered_data, function (d) {
-            return +d[context.config.value_col];
+
+        //Set x-domain based on currently selected measure.
+        this.config.x.domain = d3.extent(this.measure_data, function (d) {
```

Comment:
Don't think having this on pre-process is enough. Will only happen once ... maybe if we move to redraw()? Leaning towards just bumping #22 out of v2.0.0. It's pretty low priority for the histogram IMO.